controllers/profiles/status_enricher_dev_test.go

```diff
@@ -0,0 +1,73 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package profiles
+
+import (
+	"context"
+	"strings"
+	"testing"
+
+	openshiftv1 "github.com/openshift/api/route/v1"
+	"github.com/stretchr/testify/assert"
+	"knative.dev/pkg/apis"
+
+	apiv08 "github.com/kiegroup/kogito-serverless-operator/api/v1alpha08"
+	"github.com/kiegroup/kogito-serverless-operator/test"
+)
+
+func Test_enrichmentStatusOnK8s(t *testing.T) {
+	t.Run("verify that the service URL is returned with the default cluster name on default namespace", func(t *testing.T) {
+
+		workflow := test.GetKogitoServerlessWorkflow("../../config/samples/"+test.KogitoServerlessWorkflowSampleDevModeYamlCR, t.Name())
+		workflow.Namespace = toK8SNamespace(t.Name())
+		service, err := defaultServiceCreator(workflow)
+		client := test.NewKogitoClientBuilder().WithRuntimeObjects(workflow, service).Build()
+		obj, err := defaultDevStatusEnricher(context.TODO(), client, workflow)
+
+		reflectWorkflow := obj.(*apiv08.KogitoServerlessWorkflow)
+		assert.NoError(t, err)
+		assert.NotNil(t, obj)
+		assert.NotNil(t, reflectWorkflow.Status.Address)
+		assert.Equal(t, reflectWorkflow.Status.Address.URL.String(), "http://"+workflow.Name+"."+workflow.Namespace+".svc.cluster.local/"+workflow.Name)
+
+	})
+}
+
+func Test_enrichmentStatusOnOCP(t *testing.T) {
+	t.Run("verify that the service URL is returned with the default cluster name on default namespace", func(t *testing.T) {
+		workflow := test.GetKogitoServerlessWorkflow("../../config/samples/"+test.KogitoServerlessWorkflowSampleDevModeYamlCR, t.Name())
+		workflow.Namespace = toK8SNamespace(t.Name())
+		service, err := defaultServiceCreator(workflow)
+		route := &openshiftv1.Route{}
+		route.Name = workflow.Name
+		route.Namespace = workflow.Namespace
+		route.Spec.Host = workflow.Name + "." + workflow.Namespace + ".apps-crc.testing"
+		client := test.NewKogitoClientBuilderWithAdditionalScheme(openshiftv1.SchemeGroupVersion, route).WithRuntimeObjects(workflow, service, route).Build()
+		obj, err := devStatusEnricherForOpenShift(context.TODO(), client, workflow)
+
+		reflectWorkflow := obj.(*apiv08.KogitoServerlessWorkflow)
+		assert.NoError(t, err)
+		assert.NotNil(t, obj)
+		assert.NotNil(t, reflectWorkflow.Status.Address)
+		expectedURL := apis.HTTP(route.Spec.Host)
+		expectedURL.Path = workflow.Name
+		assert.Equal(t, reflectWorkflow.Status.Address.URL.String(), expectedURL.String())
+
+	})
+}
+
+func toK8SNamespace(testName string) string {
```

Comment:
You don't need to do this since the Fake client [won't validate the names](https://github.com/kubernetes/apimachinery/blob/8d8634786e1cdf4fd147ccf460c043b0168887bd/pkg/util/validation/validation_test.go).

Reply:
yes, in fact the issue is not on the fake client side but on the parse URL (see here https://github.com/kiegroup/kogito-serverless-operator/pull/124/files#diff-295710f7f8105f231eab1e6198fbdeec25ada99bf3be2ea551d60599c5f1bc03R54) that we are using to build the address: if the namespace is malformed it will raise an error.